init.go

```diff
@@ -200,3 +209,29 @@ func createDirectories(directories []string) error {
 	}
 	return nil
 }
+
+func shutdownPlugin() {
+	log.Info("Shutting down apidAnalytics plugin")
```

Comment:
Run through race checker. I see that you are holding on to Read lock, and then taking a write lock as well.